### PR TITLE
Add target_encoder docstring 

### PR DESCRIPTION
**Pull request by fsywang**
Addresses #11

### CODE REVIEW

File: R/target_encoder.R [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#' target_encoder
+#'
+#' This function encodes categorical variables with average target values for each category.
+#'
+#' @param X_train A tibble representing the training data set containing some categorical features/columns.
+#' @param X_test A tibble representing the test set, containing some set of categorical features/columns.
+#' @param y A numeric vector or character vector representing the target variable. 
+#'  If the objective is "binary", then the vector should only contain two unique values.
+#' @param cat_columns A character vector containing the names of the categorical columns in the tibble
+#'  that should be encoding.
```
Comment (braydentang1) [inline R/target_encoder.R]: There is a typo here, I had this originally as well. Should say "A character vector containing the names of the categorical columns in the tibble that should be en**coded**."

File: R/target_encoder.R [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#' target_encoder
+#'
+#' This function encodes categorical variables with average target values for each category.
+#'
+#' @param X_train A tibble representing the training data set containing some categorical features/columns.
+#' @param X_test A tibble representing the test set, containing some set of categorical features/columns.
+#' @param y A numeric vector or character vector representing the target variable. 
+#'  If the objective is "binary", then the vector should only contain two unique values.
```
Comment (braydentang1) [inline R/target_encoder.R:8]: Maybe mention here that the positive class should be encoded as 1, and the negative as 0 or something? What do you think?

Comment (fsywang) [reply R/target_encoder.R:8]: I think this might depend on the input. The encoder would just calculate the average regardless of the meaning of targets

Comment (braydentang1) [reply R/target_encoder.R:8]: Ah, okay that makes sense.

**Review by braydentang1 — APPROVED** [review]

**Pull request merged**